### PR TITLE
[Merged by Bors] - feat(BigOperators): prod_ite_eq_of_mem

### PR DESCRIPTION
**Pull request by amellendijk**
I found these lemmas very helpful for manipulating sums in my Selberg sieve project. 

The benefit these lemmas have over `prod_ite_eq` and `prod_ite_eq'` is that the RHS is not an if-then-else block, which makes it much more useful for rewriting backwards to introduce a sum.

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Algebra/BigOperators/Group/Finset.lean [hunk 1]
```diff
@@ -1165,6 +1165,17 @@ theorem prod_ite_eq' [DecidableEq α] (s : Finset α) (a : α) (b : α → β) :
     (∏ x ∈ s, ite (x = a) (b x) 1) = ite (a ∈ s) (b a) 1 :=
   prod_dite_eq' s a fun x _ => b x
 
+@[to_additive]
+theorem prod_ite_eq_of_mem [DecidableEq α] (s : Finset α) (a : α) (b : α → β) (h : a ∈ s) :
+    (∏ x ∈ s, if a = x then b x else 1) = b a := by
+  simp only [prod_ite_eq, h, ↓reduceIte]
+
+/-- The difference with `Finset.prod_te_eq_of_mem` is that the arguments ot `Eq` are swapped. -/
```
Comment (b-mehta) [inline Mathlib/Algebra/BigOperators/Group/Finset.lean:1173]:
```suggestion
/-- The difference with `Finset.prod_ite_eq_of_mem` is that the arguments to `Eq` are swapped. -/
```

**Review by b-mehta — COMMENTED** [review]
Can you point me to where you actually use these lemmas? The PR #20008 is what you link to, but these lemmas aren't used there.

I'm doubtful that these are actually nice use to rewrite in reverse, since you need to specify quite a bit, and I'd expect unification to struggle too.

File: Mathlib/Algebra/BigOperators/Group/Finset.lean [hunk 2]
```diff
@@ -1165,6 +1165,17 @@ theorem prod_ite_eq' [DecidableEq α] (s : Finset α) (a : α) (b : α → β) :
     (∏ x ∈ s, ite (x = a) (b x) 1) = ite (a ∈ s) (b a) 1 :=
   prod_dite_eq' s a fun x _ => b x
 
+@[to_additive]
+theorem prod_ite_eq_of_mem [DecidableEq α] (s : Finset α) (a : α) (b : α → β) (h : a ∈ s) :
+    (∏ x ∈ s, if a = x then b x else 1) = b a := by
+  simp only [prod_ite_eq, h, ↓reduceIte]
```
Comment (b-mehta) [inline Mathlib/Algebra/BigOperators/Group/Finset.lean:1171]:
```suggestion
  simp only [prod_ite_eq, if_pos h]
```
optional (but if you do like it, do the same below)

File: Mathlib/Algebra/BigOperators/Group/Finset.lean (hunk 1 above)
Comment (FLDutchmann) [reply Mathlib/Algebra/BigOperators/Group/Finset.lean:1173]: I use `sum_ite_eq_of_mem'` five times in the draft PR. Admittedly I only rewrite in reverse once [here](https://github.com/leanprover-community/mathlib4/pull/20008/files#diff-0c922162d6078c849fff74f8322f93ac418fbfb2a413cfbe1204e744472df019R211), and indeed it needs some pretty strong unification hints. I can get rid of the reverse rewrite by adding an extra line to the calc block, which may be the right thing to do here. 

That said I just attempted to replace all uses of `rw [sum_ite_eq_of_mem']` with `rw [sum_ite_eq', if_pos]` and I ran into a usability issue [here](https://github.com/leanprover-community/mathlib4/pull/20008/files#diff-0c922162d6078c849fff74f8322f93ac418fbfb2a413cfbe1204e744472df019R308). What happens is that the sum I'm removing is on the RHS of the equation, and the following `if_pos` matches with an `if then else` block on the LHS. Perhaps this is enough of a reason to keep these lemmas?

**Comment by b-mehta** [comment]
Thanks!

bors merge

**Pull request closed without merging**